Tests/NIOHTTP1Tests/HTTPDecoderTest.swift

```diff
@@ -745,4 +745,17 @@ class HTTPDecoderTest: XCTestCase {
         XCTAssertNoThrow(XCTAssert(try channel.finish().isClean))
     }
 
+    func testAppropriateErrorWhenReceivingUnsolicitedResponse() throws {
```

Comment:
could we add a test case that verifies that we also cannot silently 'recover' from having a bad server? Ie:

- `-> HTTP/1.1 200 OK\r\nServer: a-bad-server/1.0.0\r\n\r\n`
- `<- GET / HTTP/1.1`
- `-> HTTP/1.1 200 OK\r\nServer: a-bad-server/1.0.0\r\n\r\n`

isn't accepted

Reply:
I think right now it absolutely will be. Do we need to defend against that behaviour? This seems like the kind of thing we normally leave to having a pipeline error handler.

Reply:
@Lukasa I think we should protect. Not that we should close the connection but I believe we should consistently error. A server that starts responding before a request isn't quite right and I think we should continue sending errors. But happy to discuss that.

Reply:
Ok, test added to verify.